### PR TITLE
PHP 7 Support for phongo!

### PR DESCRIPTION
**Pull request by bjori**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,10 @@
 language: php
 php:
+  - '7.0'
+  - '5.6'
```
Comment (derickr) [inline .travis.yml]: Why is 5.6 in here twice? :-)

Comment (bjori) [inline .travis.yml]: Heh. Good catch, no idea. 
Travis only seems to be running 5.6 once though, so thats cool

File: phongo_compat.h [hunk 2]
```diff
@@ -112,3 +145,76 @@
 # endif
 #endif
 
+#if PHP_VERSION_ID >= 70000
+# define phongo_char zend_string
+# define phongo_char_pdup(str) pestrdup(filename->val, 1)
```
Comment (derickr) [inline phongo_compat.h]: Shouldn't this be "str->val" instead of "filename->val"?

File: phongo_compat.h [hunk 3]
```diff
@@ -112,3 +145,76 @@
...
+# define ADD_ASSOC_BOOL_EX(_zv, _key, _value) add_assoc_bool_ex(_zv, ZEND_STRL(_key), _value);
+# define phongo_free_object_arg zend_object
+# define phongo_zpp_char_len size_t
+# define ZEND_HASH_APPLY_COUNT(ht) (ht)->u.v.nApplyCount
+# define PHONGO_RETVAL_STRINGL(s, slen) RETVAL_STRINGL(s, slen)
+# define PHONGO_RETURN_STRINGL(s, slen) RETURN_STRINGL(s, slen)
+# define PHONGO_RETURN_STRING(s) RETURN_STRING(s)
+#else
+# define phongo_char char
+# define phongo_char_pdup(str) pestrdup(filename, 1)
```
Comment (jmikola) [inline phongo_compat.h]: Should `filename` be `str` here?

File: phongo_compat.h (hunk 2 above)
Comment (bjori) [inline phongo_compat.h]: I bet its only used in one place, and "filename" is the variable that is passed to it

File: php_phongo.c [hunk 4]
```diff
@@ -45,13 +45,15 @@
...
-#include <ext/date/php_date.h>
 /* Stream wrapper */
 #include <main/php_streams.h>
 #include <main/php_network.h>
 /* Debug log writing */
 #include <main/php_open_temporary_file.h>
+/* For formating timestamp in the log */
+#include <ext/date/php_date.h>
+/* String manipulation */
+#include <Zend/zend_string.h>
```
Comment (jmikola) [inline php_phongo.c]: Should this be conditional for [PHP 7](http://lxr.php.net/xref/PHP_MASTER/Zend/zend_string.h) or [PHP 5.x](http://lxr.php.net/xref/PHP_5_6/Zend/zend_string.h)? The implementations are quite different.

Comment (bjori) [inline php_phongo.c]: I don't quite see the benefit of making it conditional?
Its probably even implicitly added already, we use str_efree() from that header file, and phongo_compat.h seems to include it.... So maybe just remove it if you want to reduce the includes.

**Comment by jmikola** [comment]
Superseded by #136

**Pull request closed without merging**